tools/dep_updaters/update-icu.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/sh
+set -e
+# Shell script to update icu in the source tree to a specific version
+
+BASE_DIR=$(cd "$(dirname "$0")/../.." && pwd)
+DEPS_DIR="$BASE_DIR/deps"
+TOOLS_DIR="$BASE_DIR/tools"
+
+[ -z "$NODE" ] && NODE="$BASE_DIR/out/Release/node"
+[ -x "$NODE" ] || NODE=$(command -v node)
+
+NEW_VERSION="$("$NODE" --input-type=module <<'EOF'
+const res = await fetch('https://api.github.com/repos/unicode-org/icu/releases/latest');
+if (!res.ok) throw new Error(`FetchError: ${res.status} ${res.statusText}`, { cause: res });
+const { tag_name } = await res.json();
+console.log(tag_name.replace('release-', ''));
+EOF
+)"
+
+ICU_VERSION_H="$DEPS_DIR/icu-small/source/common/unicode/uvernum.h"
+CURRENT_MAJOR_VERSION=$(grep "#define U_ICU_VERSION_MAJOR_NUM" "$ICU_VERSION_H" | cut -d ' ' -f3 | tr -d '\r')
+CURRENT_MINOR_VERSION=$(grep "#define U_ICU_VERSION_MINOR_NUM" "$ICU_VERSION_H" | cut -d ' ' -f3 | tr -d '\r')
+
+CURRENT_VERSION="$CURRENT_MAJOR_VERSION-$CURRENT_MINOR_VERSION"
+
+echo "Comparing $NEW_VERSION with $CURRENT_VERSION"
+
+if [ "$NEW_VERSION" = "$CURRENT_VERSION" ]; then
+  echo "Skipped because icu is on the latest version."
+  exit 0
+fi
+
+NEW_MAJOR_VERSION=$(echo "$NEW_VERSION" | cut -d '-' -f1)
+NEW_MINOR_VERSION=$(echo "$NEW_VERSION" | cut -d '-' -f2)
+
+NEW_VERSION_TGZ="https://github.com/unicode-org/icu/releases/download/release-${NEW_VERSION}/icu4c-${NEW_MAJOR_VERSION}_${NEW_MINOR_VERSION}-src.tgz"
+
+./configure --with-intl=full-icu --with-icu-source="$NEW_VERSION_TGZ"
+
+"$TOOLS_DIR/icu/shrink-icu-src.py"
+
+rm -rf "$DEPS_DIR/icu"
+
+CHECKSUM=$(curl -s "$NEW_VERSION_TGZ" | md5sum | cut -d ' ' -f1)
```

Comment:
I think we should be validating this checksum matches the appropriate one from 
`https://github.com/unicode-org/icu/releases/download/release-${NEW_VERSION}/icu4c-${NEW_MAJOR_VERSION}_${NEW_MINOR_VERSION}-src.md5`

(We've had mismatches before.)

Reply:
and even check the GPG key against KEYS

Reply:
should we check the generated checksum and the one deposited match? if they dont match we skip?

Reply:
That would be my recommendation.